src/index.d.ts

```diff
@@ -0,0 +1,134 @@
+import { QueryBuilderResult } from './types';
+
+export = AEMHeadless;
+/**
+ * This class provides methods to call AEM GraphQL APIs.
+ * Before calling any method initialize the instance
+ * with GraphQL endpoint, GraphQL serviceURL and auth if needed
+ */
+declare class AEMHeadless {
+    /**
+     * Constructor.
+     *
+     * If param is a string, it's treated as AEM server URL, default GraphQL endpoint is used.
+     * For granular params, use config object
+     *
+     * @param {string|object} config - Configuration object, or AEM server URL string
+     * @param {string} [config.serviceURL] - AEM server URL
+     * @param {string} [config.endpoint] - GraphQL endpoint
+     * @param {(string|Array)} [config.auth] - Bearer token string or [user,pass] pair array
+     * @param {object} [config.headers] - header { name: value, name: value, ... }
+     * @param {object} [config.fetch] - custom Fetch instance
+     */
+    constructor(config: string | object);
+    auth: any;
+    headers: any;
+    serviceURL: string;
+    endpoint: string;
+    fetch: any;
+    /**
+     * Returns a Promise that resolves with a POST request JSON data.
+     *
+     * @param {string|object} body - the query string or an object with query (and optionally variables) as a property
+     * @param {object} [options={}] - additional POST request options
+     * @param {object} [retryOptions={}] - retry options for @adobe/aio-lib-core-networking
+     * @returns {Promise<any>} - the response body wrapped inside a Promise
```

Comment:
should it be more specific about the response type? any chance it can be different from the standard HTTP Response?

Reply:
@easingthemes wdyt about this?
let's try to close this and make a release as well